### PR TITLE
Fix a typo in `CreateCheckRun` method

### PR DESCRIPTION
**Pull request by upodroid**
/cc @alvaroaleman 

- Fixed a typo in `CreateCheckRun` method
- Added some tests
- For some reason, an empty context was appearing on the list so I added check to remove it.
- Something I forgot earlier is to block creation of a duplicate checkrun for checkruns that are successful.

https://github.com/knative/serving/pull/13176#issuecomment-1210456459
Hook Log Entry:
```
{"author":"upodroid", "component":"hook", "error":"status code 404 not one of [201], body: {"message":"Not Found","documentation_url":"https://docs.github.com/rest"}", "event-GUID":"8d96cc50-1894-11ed-90ae-a156854ac635", "event-type":"issue_comment", "file":"k8s.io/test-infra/prow/plugins/override/override.go:526", "func":"k8s.io/test-infra/prow/plugins/override.handle", "level":"warning", "msg":"cannot create prow-override CheckRun {0  64485201bcd4dc322bfdfa5b4d65b2c334827d12     completed success   {Prow override - trailing whitespace Prow has received override command for the trailing whitespace checkrun.  0  [] []} trailing whitespace {0         <nil> <nil> [] <nil>} {0   {   0  {false false false false false} }       <nil> []} []}", "org":"knative", "plugin":"override", "pr":13176, "repo":"serving", "url":"https://github.com/knative/serving/pull/13176#issuecomment-1210457420"}
```

### CODE REVIEW

File: prow/plugins/override/override.go [hunk 1]
```diff
@@ -441,11 +443,13 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent,
...
-The following unknown contexts were given:
+The following unknown contexts/checkruns were given:
+%s
+
+Only the following contexts/checkruns were expected:
 %s
 
-Only the following contexts were expected:
-%s`, formatList(unknown.List()), formatList(contexts.List()))
+Successful checkruns are listed but can not be overridden.`, formatList(unknown.List()), formatList(contexts.List()))
```
Comment (alvaroaleman) [inline prow/plugins/override/override.go:452]: Shouldn't we exclude them?

Comment (upodroid) [reply prow/plugins/override/override.go:452]: No, because succesful statuses are listed too

Comment (alvaroaleman) [reply prow/plugins/override/override.go:452]: Right and that is wrong, isn't it? Why would we do that for checkruns and not for contexts? It makes way more sense to filter those out above or not?

File: prow/plugins/override/override.go [hunk 2]
```diff
@@ -510,7 +514,7 @@ Only the following contexts were expected:
 	// Checkruns have been converted to contexts and deduped
 	if oc.UsesAppAuth() {
 		for _, checkrun := range checkrunContexts {
-			if overrides.Has(checkrun.Context) {
+			if overrides.Has(checkrun.Context) && checkrun.State != "SUCCESS" {
```
Comment (alvaroaleman) [inline prow/plugins/override/override.go:517]:
```suggestion
			if overrides.Has(checkrun.Context) && !done.Has(checkrun.Context) && checkrun.State != "SUCCESS" {
```

File: prow/plugins/override/override.go (hunk 1 above)
Comment (upodroid) [reply prow/plugins/override/override.go:452]: Ah, I can modify it to remove successful checkruns and contexts from the list that goes on the comment.

Comment (upodroid) [reply prow/plugins/override/override.go:452]: Fixed

**Review by alvaroaleman — APPROVED** [review]
/label tide/merge-method-squash

thanks!

**Pull request merged**